File: java-project-euler-100/src/main/java/net/talaatharb/projecteuler100/P022.java
package net.talaatharb.projecteuler100;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

/**
 * P022
 * 
 * @author mharb
 *
 */
public class P022 extends Solution {

	/**
	 * Problem solution entry point
	 * 
	 * @param args
	 */
	public static void main(String[] args) {
		new P022().run();
	}

	private List<String> readNames(String fileName) {
		List<String> names = new ArrayList<>();
		ClassLoader classLoader = Thread.currentThread().getContextClassLoader();

		InputStreamReader streamReader = new InputStreamReader(classLoader.getResourceAsStream(fileName));
		BufferedReader reader = new BufferedReader(streamReader);
		
		try {
			String data = reader.readLine();
			String[] parts = data.split(",");
			for (String part : parts) {
				names.add(part.substring(1, part.length() - 1));
			}
			
			Collections.sort(names);
		} catch (IOException e) {
			e.printStackTrace();
		}
		return names;
	}
	
	private int calculateNameValue(String name) {
        int value = 0;
        for(int i = 0; i < name.length(); i++) {
            value += (name.charAt(i) - ('A') + 1);
        }
        return value;
	}

	@Override
	public Number solve() {
		problemNumber = 22;
		int result = 0;
		List<String> names = readNames("p022.txt");
		for(int i = 0; i < names.size(); i++) {
			result += calculateNameValue(names.get(i)) * (i+1);
		}

		return result;
	}

}

File: typescript-project-euler-100/src/p022.ts
import { readFileSync } from "fs";
import { Solution } from "./Solution";

export class P022 extends Solution {

	readNames(fileName: string): string[] {
		const data = readFileSync(fileName, "utf-8");
		const names = data.split(",").map(part => part.slice(1, -1));
		names.sort();
		return names;
	}

	calculateNameValue(name: string): number {
		let value = 0;
		for (const letter of name) {
			value += letter.charCodeAt(0) - "A".charCodeAt(0) + 1;
		}
		return value;
	}

    solve() {
        this.problemNumber = 22;
		const names = this.readNames("p022.txt");
		let result = 0;
		for (let i = 0; i < names.length; i++) {
			result += this.calculateNameValue(names[i]) * (i + 1);
		}

		return result;
    }
}
